alacritty/src/display/mod.rs

```diff
@@ -552,9 +558,59 @@ impl Display {
         }
     }
 
-    pub fn make_current(&self) {
-        if !self.context.get().is_current() {
-            self.context.make_current(&self.surface).expect("failed to make context current")
+    pub fn make_current(&mut self, recover: bool) {
+        let is_current = self.context.get().is_current();
+        if is_current && !recover {
+            return;
+        }
+
+        let mut was_context_reset = if is_current {
+            false
+        } else {
+            match self.context.make_current(&self.surface) {
+                Err(err) if err.error_kind() == ErrorKind::ContextLost => {
+                    info!("Context lost for window {:?}", self.window.id());
+                    true
+                },
+                _ => false,
+            }
+        };
+
+        was_context_reset |= self.renderer.was_context_reset();
+
+        if was_context_reset && recover {
+            let gl_display = self.context.display();
```

Comment:
This is a pretty big chunk of code inside this conditional. Could probably move it to a function or early-return.